src/Devices/PushingBox.cpp

```diff
@@ -0,0 +1,65 @@
+#include "PushingBox.h"
+
+#include "Devices/EEPROM_TC.h"
+#include "Devices/Ethernet_TC.h"
+#include "Devices/PHProbe.h"
+#include "Devices/Serial_TC.h"
+#include "Devices/TempProbe_TC.h"
+
+//  class variables
+PushingBox* PushingBox::_instance = nullptr;
+
+//  class methods
+/**
+ * accessor for singleton
+ */
+PushingBox* PushingBox::instance() {
+  if (!_instance) {
+    _instance = new PushingBox();
+    Ethernet_TC::instance();
+  }
+  return _instance;
+}
+
+void PushingBox::loop() {
+  // is it time to send ?
+  unsigned long now = millis();
+  if (now >= nextSendTime) {
+    sendData();
+    unsigned long interval = EEPROM_TC::instance()->getGoogleSheetInterval();
+    // jump to the next multiple of interval
+    nextSendTime = (now / interval + 1) * interval;
+  }
+  // are we still connected?
+  if (client.connected()) {
+    // if so, read response
+    int next;
+    while ((next = client.read()) != -1) {
```

Comment:
Codacy "is a very simple pattern-matching tool, and all it cares about there is that the read function is being used - it doesn't parse the arguments or do any semantic analysis. It's not really intended for use without a human looking over its shoulder." [cite](https://github.com/jamulussoftware/jamulus/issues/1081). I'm not sure how we should handle this. Perhaps there is a way to suppress the warning for a particular line. Or, we could just accept the PR and Codacy won't complain again until the line is edited.